### PR TITLE
[stable/weave-cloud] Replace Scope, Flux and Cortex with weave-agent daemon

### PR DESCRIPTION
**Pull request by stefanprodan**
**What this PR does / why we need it**:

- replace Scope, Flux and Cortex with weave-agent daemon
- use the `weave` namespace instead of `kube-system`
- rename the token variable and store it as a kube secret
- bump chart version to 0.2.0

**Special notes for your reviewer**:

Tested on GKE v1.8.9 and v1.9.6

### CODE REVIEW

**Review by errordeveloper — COMMENTED** [review]
I will review this once we have sorted out the CLA issue. I believe Stefan is covered by Weaveworks CLA already.

**Comment by stefanprodan** [comment]
CLA sorted I hope 😀

File: stable/weave-cloud/README.md [hunk 1]
```diff
@@ -44,4 +50,4 @@ The following tables lists the configurable parameters of the Weave Cloud Agents
 
 | Parameter | Description | Default |
 | --------- | ----------- | ------- |
-| `ServiceToken` | Weave Cloud service token | _none_ _(**must be set**)_ |
+| `token` | Weave Cloud service token | _none_ _(**must be set**)_ |
```
Comment (errordeveloper) [inline stable/weave-cloud/README.md:56]: I'm okay with changing this, as don't think many people use this chart.

File: stable/weave-cloud/templates/NOTES.txt [hunk 2]
```diff
@@ -1,10 +1,10 @@
-{{- if .Values.ServiceToken -}}
+{{- if .Values.token -}}
```
Comment (errordeveloper) [inline stable/weave-cloud/templates/NOTES.txt:1]: You will need to change the logic here to check for the default value.

File: stable/weave-cloud/values.yaml [hunk 3]
```diff
@@ -1 +1,9 @@
-# ServiceToken: ""
+# Default values for weave-cloud.
+
+token: demo
```
Comment (errordeveloper) [inline stable/weave-cloud/values.yaml]: I'd make it something else, e.g. `x` and leave a comment saying that this has to be set to something in oder for e2e tests to pass.

File: stable/weave-cloud/templates/NOTES.txt (hunk 2 above)
Comment (stefanprodan) [reply stable/weave-cloud/templates/NOTES.txt:1]: If I'm checking for the default value the e2e tests will fail 😢

Comment (errordeveloper) [reply stable/weave-cloud/templates/NOTES.txt:1]: interesting... it sounds like we should really get someone who knows more about e2e tests to help us!

**Comment by errordeveloper** [comment]
I am not sure how we could make things work properly here.

The problem is this the agent requires a token if it is given an empty string it crashes, which is by design. So to make the chart pass e2e test, we have set default value of the token to `demo`, but that means we now have a default value that suppresses warning we produce via [`NOTES.txt`](https://github.com/stefanprodan/charts/blob/4dd0ea2eed094461ab4abf7d4cdfcb6f8392fdfb/stable/weave-cloud/templates/NOTES.txt). @stefanprodan said that for some reason e2e test fails when we try to check the default value in `NOTES.txt`... I'm puzzled. Is there a way to provide values file for e2e test to use or pass custom args to `helm install` in e2e run?

**Comment by unguiculus** [comment]
Please have a look at our guidelines especially regarding resource names and labels.

https://github.com/kubernetes/charts/blob/master/REVIEW_GUIDELINES.md

Please follow our RBAC best practices.

https://github.com/kubernetes/helm/blob/master/docs/chart_best_practices/rbac.md

Custom test values are possible.

https://github.com/kubernetes/charts/blob/master/test/README.md#providing-custom-test-values

**Comment by stefanprodan** [comment]
@unguiculus thanks for the help, I've used the CI dir.

File: stable/weave-cloud/templates/NOTES.txt (hunk 2 above)
Comment (stefanprodan) [reply stable/weave-cloud/templates/NOTES.txt:1]: Using the CI dir solved this, the default values contain an empty token now.

File: stable/weave-cloud/Chart.yaml [hunk 4]
```diff
@@ -1,10 +1,14 @@
+apiVersion: v1
+appVersion: "0.2.0"
 name: weave-cloud
-version: 0.1.3
+version: 0.2.0
 description: |
     Weave Cloud is a add-on to Kubernetes which provides Continuous Delivery, along with hosted Prometheus Monitoring and a visual dashboard for exploring & debugging microservices
 home: https://weave.works
 maintainers:
   - name: Ilya Dmitrichenko
```
Comment (unguiculus) [inline stable/weave-cloud/Chart.yaml]: Please use Github usernames.

File: stable/weave-cloud/ci/e2e-values.yaml [hunk 5]
```diff
@@ -0,0 +1,13 @@
+# CI values for weave-cloud.
+
+token: demo
```
Comment (unguiculus) [inline stable/weave-cloud/ci/e2e-values.yaml:3]: The only thing that changes for CI seems to be the token. Don't duplicate the rest.

File: stable/weave-cloud/templates/deployment.yaml [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+      {{- if .Values.serviceAccount.create }}
+      serviceAccount: {{ .Values.agent.name }}
+      serviceAccountName: {{ .Values.agent.name }}
+      {{- end }}
+      terminationGracePeriodSeconds: 30
+      containers:
+        - name: {{ .Values.agent.name }}
+          image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          args:
```
Comment (unguiculus) [inline stable/weave-cloud/templates/deployment.yaml]: Please use consistent indentation style throughout the chart. Either zero or two spaces.

File: stable/weave-cloud/values.yaml [hunk 7]
```diff
@@ -1 +1,19 @@
...
+# Default values for weave-cloud.
+
+token: ""
+
+agent:
+  name: weave-agent
+
+image:
+  repository: quay.io/weaveworks/launcher-agent
+  tag: master-b60524c
```
Comment (unguiculus) [inline stable/weave-cloud/values.yaml]: Please use a release version.

**Review by unguiculus — CHANGES_REQUESTED** [review]
Please read up on RBAC and best practices again:

* There should be a partial for the service account in `_helpers.tpl`.
* Resource names should be based on the `fullname` partial.

**Comment by stefanprodan** [comment]
@unguiculus thanks for the review, can you please take a look again. I think I've addressed all of your remarks.

File: stable/weave-cloud/templates/deployment.yaml [hunk 8]
```diff
@@ -13,30 +13,30 @@ spec:
...
     metadata:
       labels:
-        app: {{ .Values.agent.name }}
+        app: {{ template "weave-cloud.name" . }}
         release: {{ .Release.Name }}
     spec:
       {{- if .Values.serviceAccount.create }}
-      serviceAccount: {{ .Values.agent.name }}
-      serviceAccountName: {{ .Values.agent.name }}
+      serviceAccount: {{ template "weave-cloud.serviceAccountName" . }}
```
Comment (unguiculus) [inline stable/weave-cloud/templates/deployment.yaml]: You shouldn't need this. It's deprecated.

**Review by unguiculus — CHANGES_REQUESTED** [review]

File: stable/weave-cloud/templates/deployment.yaml (hunk 8 above)
Comment (stefanprodan) [reply stable/weave-cloud/templates/deployment.yaml]: I've removed it. Thanks

**Review by unguiculus — APPROVED** [review]

**Pull request merged**